packages/block-library/src/image/edit.native.js

```diff
@@ -722,6 +722,8 @@ export class ImageEdit extends Component {
 			);
 		}
 
+		const isGif = url.endsWith( '.gif' );
```

Comment:
Separate to the point Siobhan made above, I wonder if we'd like to make the condition case-insensitive so filenames ending in, say,  `.Gif` could also be considered a GIF.

Come to think of it, should we perhaps have a different way (or fallback perhaps) for detecting GIFs? I tried using a GIF that I intentionally modified its extension to not big `gif` at all and looks like Gutenberg-web can accept and render it fine. It'd be cool if we could add the badge to such file, wdyt? Try the following block in https://wordpress.org/gutenberg/:

```
<!-- wp:image {"sizeSlug":"large"} -->
<figure class="wp-block-image size-large"><img src="https://cldup.com/Up7yfYlqFj.tzif" alt=""/></figure>
<!-- /wp:image -->
```

Reply:
Interesting idea. I think I can look at the image itself -- using `DataView` to see if there is a delay time in the image.

Reply:
> using DataView

and since I noticed the https://github.com/WordPress/gutenberg/pull/38996/commits/a454427703cd3f7ff0f24b41012cc8816fa8ac71 that includes code to read the raw data of the image, I wonder if that's a function we can depend on a 3rd party library to perform. Not sure, maybe you have already researched and dismissed such option @illusaen ? I see https://www.npmjs.com/package/animated-gif-detector, not sure if that could fit the purpose.

Reply:
So I tried using a package but it's rather more complicated than expected as it requires `fs`. Skipping for `.includes` for now.

Reply:
👋 Wendy! Noticed that with https://github.com/WordPress/gutenberg/pull/38996/commits/25ed3d5158c9e6dbeba0932fdfb9bfad1cb1a6ac, the dot is removed from the filename check. Can you elaborate on that choice? Won't that mean that any occurrence of the `gif` substring will be detected as gif, regardless if it's in the filename extension or not? 🤔

Reply:
To be honest, that's an oversight, not a deliberate choice. Fixed!

Reply:
Kinda a nit but it touches a few places so I thought is better to mention it: the flag is named `isAnimatedGif` currently, but I think the simpler/general `isGif` is more accurate since we are not really detecting whether the GIF is really animated or not (GIFs can be static too). The "Animated" moniker was introduced in a454427703cd3f7ff0f24b41012cc8816fa8ac71 but we since have removed that logic so, wdyt about going back to the simple `isGif`?

We should also remove the "Animated" qualifier from the release notes as well and only mention GIF.